### PR TITLE
Add component place-detail

### PR DESCRIPTION
**Pull request by helenrivne**

### CODE REVIEW

File: client/components/place/place-detail/place-detail.component.js [hunk 1]
```diff
@@ -4,23 +4,30 @@ angular.module('placeDetail', [])
...
-      this.location = this.place.getLocation();
-      this.marker = L.marker(this.location).addTo(this.map);
+      this.marker = L.marker(L.latLng(this.place.location.coordinates[0], this.place.location.coordinates[1])).addTo(this.map);
+      this.marker.bindPopup('<div class="popup"><h3>' + this.place.name +
+        '</h3><a><img class="marker-image center-block" src="'+this.place.photos[0]+'" /></a><br />').openPopup();
 
-      this.marker.bindPopup('<div><h3>' + this.place.name +
-        '</h3><a><img class="marker-image" src="assets/' + this.place.photo[0] +
-        '" /></a><br />').openPopup();
+      this.map.setView([this.place.location.coordinates[0]+0.005, this.place.location.coordinates[1]]); //0.005- for responcive design- show info on mobile
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Magic numbers :8ball:

File: client/components/place/place-detail/place-detail.component.js [hunk 2]
```diff
@@ -4,23 +4,30 @@ angular.module('placeDetail', [])
...
+        document.getElementById("file1").click();
+      }
+
       this.map = L.map('map1', {
-        center: [50.6234, 26.2189],
-        zoom: 14
+        center: constants.mapCenter,
+        zoom: constants.defaultZoom
       });
       Streets = L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Що це за глобальна змінна `Streets` тут?

File: client/components/place/place-detail/place-detail.template.html [hunk 3]
```diff
@@ -1,73 +1,93 @@
...
-  <h3>Place:{{$ctrl.place.id}}</h3>
-
-
-  <div class="col-lg-3 col-md-3 col-xs-12 " style="overflow: hidden">
-    <div class="row">
-      <a class="fancybox" rel="gallery1" href="assets/{{$ctrl.place.photo[0]}}" title="{{$ctrl.place.name}}">
-        <img src="assets/{{$ctrl.place.photo[0]}}" alt="" style="min-width:300px; max-width:600px; width:400px;"
-        />
-      </a>
+  <div class="col-xs-12 col-lg-12 col-md-12" style="min-width:365px;">
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Що це тут за інлайн стилі?

File: client/components/place/place-detail/place-detail.template.html [hunk 4]
```diff
@@ -1,73 +1,93 @@
...
-            <textarea class="form-control" id="comment" rows="3"></textarea>
-          </div>
+      <div class="row vertical-align">
+        <div class="pull-left"><img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="profile"/>
+        </div>
+        <div class="col-xs-8">
+          <form class="form-inline">
+            <div class="form-group" class="col-xs-11">
+              <label class="sr-only" for="comment">Email address</label>
+                            <textarea class="form-control col-xs-9" id="comment" rows="4" cols="70"
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Тут `textarea` з'їхав

File: client/components/place/place-detail/place-detail.template.html [hunk 5]
```diff
@@ -1,73 +1,93 @@
...
+                                      placeholder="Leave your comments for other..."></textarea>
+            </div>
+            <div class="form-group">
+              <i class="fa fa-camera-retro fa-3x" onclick="upload()"
+                 onmouseover="this.style.color='lightgreen'" onmouseout="this.style.color='#3da268'"
+                 id="upfile1"></i>
+              <input type="file" class="btn btn-default form-control" id="file1" style="display:none"/>
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Інлайн стилі?

File: client/components/place/place-detail/place-detail.template.html [hunk 6]
```diff
@@ -1,73 +1,93 @@
...
-          <button type="submit" class="btn btn-default">Add Comment</button>
-        </form>
-        <br>
-        <br>
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
         </div>
+      </div>
+      <div class="comment">
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Отут під коменти Макса треба переробить

File: client/components/place/place-detail/place-detail.template.html [hunk 7]
```diff
@@ -1,73 +1,93 @@
...
-    <div class="col-xs-12 no-padding">
-      <!-- Map container start-->
-      <div id="map1"></div>
-    </div>
-  </div>
+    <div class="col-lg-4 col-md-4 col-xs-12 galery">
+      <div class="row">
+        <a class="fancybox" rel="gallery1" href="{{$ctrl.place.photos[0]}}" title="{{$ctrl.place.name}}">
+          <img src="{{$ctrl.place.photos[0]}}" alt="{{$ctrl.place.name}}"
+               style="width:420px;"/>
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Інлайн стилі

File: client/app.config.js [hunk 8]
```diff
@@ -5,3 +5,23 @@ angular.module('greenTourism')
       id: '_id'
     });
   }]);
+
+angular.module('greenTourism')
+  .controller('MainCtrl', ['$rootScope', '$scope', function MainCtrl($rootScope, $scope) {
```
Comment (yurykozhenov) [inline client/app.config.js]: I moved `MainCtrl` into seperate file, move all your changes there and remove them from here

File: client/assets/less/pages/place-detail.less [hunk 9]
```diff
@@ -37,14 +37,17 @@
 div.place {
   margin: 0px;
   .vertical-align {
-    padding: 5px 10px;
+
+    padding: 10px 5px 0px 5px;
```
Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less]: Change `0px` to `0`

File: client/assets/less/pages/place-detail.less [hunk 10]
```diff
@@ -63,22 +66,38 @@ div.place {
...
+        display: inline-block;
+        width: 32%;
+      }
+      a img {
+        float: left;
+        width: 100%;
+        height: auto;
+        padding-bottom: 5px;
+      }
+      #firstimg {
```
Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less:88]: Do you really need id there? You can use pseudo-selector `img:first`

**Comment by yurykozhenov** [comment]
Why you added image?

File: client/assets/less/pages/place-detail.less (hunk 10 above)
Comment (helenrivne) [inline client/assets/less/pages/place-detail.less:88]: Yes, before I tried to use "a  img:first-of-type" but it didn't work

File: client/assets/less/pages/place-detail.less (hunk 9 above)
Comment (helenrivne) [inline client/assets/less/pages/place-detail.less]: Even on w3.org use 0px

Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less]: https://google.github.io/styleguide/htmlcssguide.xml?showone=0_and_Units#0_and_Units

Comment (helenrivne) [inline client/assets/less/pages/place-detail.less]: Ok, you've won)

**Comment by yurykozhenov** [comment]
Don't add image to repository. For demonstration purposes please use url on remote image source.

File: client/components/place/place-detail/place-detail.template.html [hunk 11]
```diff
@@ -1,73 +1,73 @@
...
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
-        </div>
-        <p class="clearfix">Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1
-          Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1</p>
+        <!--
+          <a ng-repeat="photo in $ctrl.place.photos" ng-if="$index>=1" class="fancybox" rel="gallery1"
+             href="{{photo}}" title="{{$ctrl.place.name}}">
+            <img src="assets/{{photo}}" alt="{{$ctrl.place.name}}"/>
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html:41]: Use `ng-src`

File: client/components/place/place-detail/place-detail.template.html [hunk 12]
```diff
@@ -1,73 +1,73 @@
...
-        <br>
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
-        </div>
-        <p class="clearfix">Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1
-          Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1</p>
+        <!--
+          <a ng-repeat="photo in $ctrl.place.photos" ng-if="$index>=1" class="fancybox" rel="gallery1"
+             href="{{photo}}" title="{{$ctrl.place.name}}">
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html:40]: Use `ng-href`

Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: What is `title`?

File: client/components/place/place-detail/place-detail.template.html [hunk 13]
```diff
@@ -1,73 +1,73 @@
...
+            <img src="assets/{{photo}}" alt="{{$ctrl.place.name}}"/>
+        </a>
+        <!-- to delete- it is for demonstrate without data in DB -->
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Delete this, we have data in DB

**Comment by yurykozhenov** [comment]
untitled

There is strange lines on your map. And also, I can't drag the map.

**Comment by yurykozhenov** [comment]
And when I press X button page doesn't close :confused:

File: client/assets/less/pages/place-detail.less [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+        width: 100%;
+        height: auto;
+        max-height: 100%;
+        border: none;
+        display: block;
+
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Delete empty rows

File: client/components/place/place-detail/place-detail.component.js [hunk 15]
```diff
@@ -1,26 +1,43 @@
-angular.module('placeDetail', [])
+angular.module('placeDetail', ['comment'])
   .component('placeDetail', {
     templateUrl: 'components/place/place-detail/place-detail.template.html',
     bindings: {
       place: '<'
     },
-    controller: ['mapFactory', function placeDetailCtrl(mapFactory) {
+    controller: function placeDetailCtrl($scope, constants) {
+      $(document).ready(function() {
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.component.js]: Change $ on angular.element

File: client/shared/search/search.template.html [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<div class="col-xs-12" id="search">
+  <div class="input-group col-xs-offset-1 col-xs-10">
+    <input class="form-control" type="text" name="search">
+          <span class="input-group-btn">
```
Comment (max-yushchuk) [inline client/shared/search/search.template.html:8]: Change 4 spaces on 2 spaces

File: client/assets/js/videoplayer.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
+angular.element(document).ready(function () {
```
Comment (max-yushchuk) [inline client/assets/js/videoplayer.js:1]: Remove this file

File: client/assets/js/videoplayer.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+  var controls = {
+    video: angular.element("#myvideo")
```
Comment (max-yushchuk) [inline client/assets/js/videoplayer.js:3]: Rename #myvideo on something like #welcome-page-video, because it is not clearly

File: client/assets/less/pages/place-detail.less [hunk 19]
```diff
@@ -0,0 +1,132 @@
+.page-detail {
+  bottom: 0;
+  padding-top: 50px;
+  position: absolute;
+  width: 100%;
+  top: 0;
+  background-color:@page-bg-color;
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Add space between : and @

File: client/assets/less/pages/place-detail.less [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+@keyframes slideInOut {
+  from {
+    transform: translateY(0%);
+  }
+  to {
+    transform: translateY(100%);
+  }
+}
+.ng-enter       { z-index: 8888; }
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Remove spaces

File: client/assets/less/pages/place-detail.less [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+.ng-leave       { z-index: 9999; }
+
+.page-detail.ng-enter {
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less:28]: Use LESS style:

``` less
.page-detail {
  .ng-enter {}
  .ng-leave
}
```

File: client/assets/less/pages/place-detail.less [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+.page-detail.ng-leave {
+  animation: slideInOut 0.5s both ease-in;
+}
+
+div.place {
+  margin: 0;
+  .vertical-align {
+
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less:41]: Remove line break

File: client/assets/less/pages/place-detail.less [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+  button .glyphicon-heart, i.close {
+    color: @brand-dark;
+  }
+  h2 {
+    color: @brand-dark;
+    text-transform: capitalize;
+    margin-top: -10px;
+  }
+  .container_page {
+    padding: 20px 20px;
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Change on padding: 20px

File: client/app.module.js [hunk 24]
```diff
@@ -6,5 +6,6 @@ angular.module('greenTourism', [
   'place',
   'track',
   'event',
-  'blog'
+  'blog',
+  'ngAnimate'
```
Comment (yurykozhenov) [inline client/app.module.js]: Move `ngAnimate` on top of file

File: client/assets/less/pages/place-detail.less [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+    div .vertical-align {
+      margin-right: 20px;
+    }
+    textarea {
+      border: 1px solid @brand-dark;
+    }
+    button#post {
+      border-radius: 0px;
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Remove "px"

File: client/assets/less/pages/place-detail.less [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+        width: 100%;
+        height: auto;
+        max-height: 100%;
+        border: none;
+        display: block;
+      }
+
+    }
+  }
+  @keyframes closeMap {
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: Remove this. It is old

File: client/assets/less/pages/place-detail.less [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+  }
+  i#close, i.fa {
+    cursor: pointer;
+  }
+
+  .closeMap {
+    position: absolute;
+    top: 10px;
+    left: 50%;
+    z-index: 10000000;
```
Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less]: Make `z-index` smaller :laughing:

File: client/assets/less/pages/welcome-page.less [hunk 28]
```diff
@@ -1,18 +1,30 @@
...
+  #myvideo {
+    max-width: 100%;
+    margin: 0 auto 0 auto;
+    right: 0;
+    bottom: 0;
+    min-width: 100%;
+    min-height: 100%;
+    width: auto;
+    height: auto;
+    z-index: -9999;
```
Comment (max-yushchuk) [inline client/assets/less/pages/welcome-page.less]: Maybe 0?

File: client/components/place/place-detail/place-detail.component.js [hunk 29]
```diff
@@ -1,26 +1,28 @@
-angular.module('placeDetail', [])
+angular.module('placeDetail', ['comment'])
   .component('placeDetail', {
     templateUrl: 'components/place/place-detail/place-detail.template.html',
     bindings: {
       place: '<'
     },
-    controller: ['mapFactory', function placeDetailCtrl(mapFactory) {
+    controller: function placeDetailCtrl($scope, constants) {
+      angular.element(document).ready (function() {
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Remove space before `(`

File: client/components/place/place-detail/place-detail.component.js [hunk 30]
```diff
@@ -1,26 +1,28 @@
...
+        $('.fancybox').fancybox();
+      });
       this.map = L.map('map1', {
-        center: [50.6234, 26.2189],
-        zoom: 14
+        center: constants.mapCenter,
+        zoom: constants.defaultZoom
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.component.js]: Add dragging stop and zoom stop

File: client/assets/less/pages/place-detail.less (hunk 19 above)
Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less]: You don't need `background-color`, it's white by default.

File: client/assets/less/variables.less [hunk 31]
```diff
@@ -2,10 +2,12 @@
 @light-green: #009000;  // TODO: Rename this
 
 @footer-bg-color: #3c3c3d;
+@page-bg-color: #ffffff;
```
Comment (yurykozhenov) [inline client/assets/less/variables.less]: Remove this variable

File: client/components/place/places-location/locationPlaces.component.js [hunk 32]
```diff
@@ -1,30 +1,50 @@
...
+
+    $scope.placesByLocation = placesArr.filter(function(place) {
+        latitude = place.location.coordinates[1];
+        longitude = place.location.coordinates[0];
+        return (latitude > min.lat && longitude > min.lng
+          && latitude < max.lat && longitude < max.lng) && place.photos[0];
       }
-    }
+    );
+    $scope.placesByLocation = _.sortBy($scope.placesByLocation, ['rate']);
```
Comment (yurykozhenov) [inline client/components/place/places-location/locationPlaces.component.js]: Why do you use `sortBy` inside of controller?
All of your places are sorting inside `ngRepeat` in template throught filter.

File: client/components/place/place-detail/place-detail.template.html [hunk 33]
```diff
@@ -1,73 +1,75 @@
...
+        <comment input-object="$ctrl.place" input-object-type="'places'"></comment>
 
-        <form>
-          <div class="form-group">
-            <label for="comment">Comment</label>
-            <textarea class="form-control" id="comment" rows="3"></textarea>
-          </div>
+      </div>
+      <div class="col-lg-4 col-md-4 col-xs-12 gallery">
+        <a id="firstimg" class="fancybox" rel="gallery1" ng-href="{{$ctrl.place.photos[0]}}">
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.template.html:34]: Remove `id="firstimg"` and use:

``` less
.fancybox {
  img:first-child
}
```

File: client/components/place/place-detail/place-detail.template.html [hunk 34]
```diff
@@ -1,73 +1,75 @@
...
-        </form>
-        <br>
-        <br>
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
-        </div>
-        <p class="clearfix">Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1
-          Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1</p>
+        <!--
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.template.html:38]: Why is this here?

File: client/components/place/place-detail/place-detail.template.html [hunk 35]
```diff
@@ -1,73 +1,75 @@
...
+          <a ng-repeat="photo in $ctrl.place.photos" ng-if="$index>=1" class="fancybox" rel="gallery1"
+             href="{{photo}}" title="{{$ctrl.place.name}}">
+            <img src="assets/{{photo}}" alt="{{$ctrl.place.name}}"/>
+        </a>
+        <!-- to delete- it is for demonstrate without data about other photo in DB -->
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.template.html:43]: If "to delete", then Do this :)

File: client/components/place/placesOnMap.factory.js [hunk 36]
```diff
@@ -40,17 +40,20 @@ angular.module('mapModule')
...
         });
         mainGroup.checkIn(groups[input]);
         groups[input].addTo(map);
       } else {
         places.forEach(function(place) {
           marker(place.location.coordinates[0], place.location.coordinates[1], types[place.type].icon)
             .addTo(groups[place.type])
-            .bindPopup('<div class=\'popup  center-block\'><h3>' + place.name + '</h3><a><img class=\'marker-image\' src=\'assets/' + place.photos[0] + '\' \/></a>' +
-              '<br /><br /><button type=\'button\' class=\'btn btn-default btn-md center-block\'> <a href=\'#!/places/' + place._id + '\'>Details >></a> </button></div>', {autoPan: false});
+            .bindPopup("<div class='popup'><h3>" + place.name + "</h3><a href='#!/places/" + place._id + "'>" +
```
Comment (yurykozhenov) [inline client/components/place/placesOnMap.factory.js]: Swap `"` and `'`

Comment (yurykozhenov) [inline client/components/place/placesOnMap.factory.js]: Line exceeds max length of 80 characters, please make it smaller

File: client/components/welcome-page/welcome-page.template.html [hunk 37]
```diff
@@ -106,3 +115,24 @@
     </div>
   </div>
 </footer>
+<script>
```
Comment (yurykozhenov) [inline client/components/welcome-page/welcome-page.template.html]: Remove script inside html

File: client/components/place/place-list/place-list.component.js [hunk 38]
```diff
@@ -180,6 +180,7 @@ angular.module('placeList', ['filterMapType', 'popularTracks', 'ngAnimate'])
             Place.getList({type: input, limit: 100}).then(function(result) {
               places = result.concat(places);
               placesOnMap.showPlaces(result, input);
+
```
Comment (max-yushchuk) [inline client/components/place/place-list/place-list.component.js:387]: Remove line break

File: client/components/welcome-page/welcome-page.template.html [hunk 39]
```diff
@@ -1,7 +1,16 @@
...
+  <div class="parallax-background-opacity">
+    <video id="myvideo" autoplay poster="http://rvavto.com.ua/images/main.jpg" preload>
+      <source src="http://rvavto.com.ua/images/video1.mp4" type="video/mp4"/>
+    </video>
+  </div>
   <div class="parallax-title">
-    <h2>Find. Plan. Travel</h2>
+    <h2>Find. Plan. Travel </h2>
+
+    <p>
```
Comment (max-yushchuk) [inline client/components/welcome-page/welcome-page.template.html]: Remove p

File: client/shared/search/search.component.js [hunk 40]
```diff
@@ -0,0 +1,15 @@
+angular.module('greenTourism')
+  .component('searchPlace', {
+    templateUrl: 'shared/search/search.template.html',
+    controller: ['Search', function SearchCtrl(Search) {
+
+      this.searchPlace = function(searchname) {
+        var searchBy = 'place';
+        //Place.getList({name: [input. name]}).then(function (result)
+        Search.getList({name: [searchname], searchBy: [searchBy]}).then(function(result) {
+          console.log("SEARCH");
```
Comment (yurykozhenov) [inline client/shared/search/search.component.js]: Remove logs

File: server/config/express.js [hunk 41]
```diff
@@ -42,8 +42,8 @@ module.exports = function() {
   api.use('/places', require('../app/routes/places'));
   api.use('/users', require('../app/routes/users'));
   api.use('/tracks', require('../app/routes/tracks'));
-  api.use('/blogs', require('../app/routes/blogs'));
```
Comment (SergiiSemenkiv) [inline server/config/express.js:45]: :hankey: Thanks for my demo

File: server/app/controllers/search.js [hunk 42]
```diff
@@ -0,0 +1,27 @@
...
+
+  //var name= req.query.name;
+  var regex = new RegExp(req.query.name, 'i');
+  delete req.query.name;
+
+  var searchBy = req.query.searchBy;
+  delete req.query.searchBy;
+
+  if (searchBy == 'place') {
+    Search.find({name: regex},
```
Comment (yurykozhenov) [inline server/app/controllers/search.js]: What is `Search`?

File: server/app/controllers/search.js [hunk 43]
```diff
@@ -0,0 +1,27 @@
...
+      function(err, records) {
+        if (err) {
+          res.status(400).json(err);
+        } else {
+          res.json(records);
+        }
+      });
+  } else {
```
Comment (yurykozhenov) [inline server/app/controllers/search.js]: Remove empty `else` block

File: server/app/controllers/search.js [hunk 44]
```diff
@@ -0,0 +1,27 @@
+var Search = require('mongoose').model('Place');
```
Comment (yurykozhenov) [inline server/app/controllers/search.js:1]: You shouldn't create separate controller for search, use `controllers/places.js` instead.

File: server/app/routes/search.js [hunk 45]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
```
Comment (yurykozhenov) [inline server/app/routes/search.js:1]: You don't need separate routes for search

**Comment by yurykozhenov** [comment]
Please, add description of all work you done in this pull request.

File: client/components/place/places-location/locationPlaces.component.js (hunk 32 above)
Comment (helenrivne) [inline client/components/place/places-location/locationPlaces.component.js]: At first I take  sorted array from bd but ondrag map I recieve array from factory

File: client/components/place/place-detail/place-detail.template.html (hunk 33 above)
Comment (helenrivne) [inline client/components/place/place-detail/place-detail.template.html:34]: It's will not work. Even img:first-of-type not working (Ho knows, why(()

File: client/components/place/place-detail/place-detail.template.html (hunk 35 above)
Comment (helenrivne) [inline client/components/place/place-detail/place-detail.template.html:43]: It's images for demo, I use it while there is not images urls in BD

File: client/components/place/places-location/locationPlaces.component.js (hunk 32 above)
Comment (yurykozhenov) [inline client/components/place/places-location/locationPlaces.component.js]: You can still leave all sorting inside `ngRepeat` and remove it from controller.
Or does this `sortBy` have some other purpose?

Comment (helenrivne) [inline client/components/place/places-location/locationPlaces.component.js]: Yes, it's no nesseserity in sorting inside filter, sorting (_.sortBy) is for taking only 5 rating places ( it's not usefull right now but if we'll do preloading or smth else, it would be usefull). I think, it no good getting in scope very big array

Comment (yurykozhenov) [inline client/components/place/places-location/locationPlaces.component.js]: You have full array in scope anyway. So there is no point of sorting and later slicing it, you're just adding complexity inside your component.
So use full array as it is or change request to server.

File: client/components/place/place-detail/place-detail.template.html (hunk 35 above)
Comment (helenrivne) [inline client/components/place/place-detail/place-detail.template.html:43]: Just DO IT!!!)))

Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html:43]: DO IT

**Comment by yurykozhenov** [comment]
You accidentally merged `demo3` in your branch...
